mr/src/main/java/org/apache/iceberg/mr/mapred/IcebergSchemaToTypeInfo.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.hadoop.hive.serde2.SerDeException;
+import org.apache.hadoop.hive.serde2.typeinfo.HiveDecimalUtils;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+/**
+ * Class to convert Iceberg types to Hive TypeInfo
+ */
+final class IcebergSchemaToTypeInfo {
+
+  private IcebergSchemaToTypeInfo() {
+
+  }
+
+  private static final ImmutableMap<Object, Object> primitiveTypeToTypeInfo = ImmutableMap.builder()
+      .put(Types.BooleanType.get(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.BOOLEAN_TYPE_NAME))
+      .put(Types.IntegerType.get(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.INT_TYPE_NAME))
+      .put(Types.LongType.get(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.BIGINT_TYPE_NAME))
+      .put(Types.FloatType.get(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.FLOAT_TYPE_NAME))
+      .put(Types.DoubleType.get(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.DOUBLE_TYPE_NAME))
+      .put(Types.BinaryType.get(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.BINARY_TYPE_NAME))
+      .put(Types.StringType.get(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.STRING_TYPE_NAME))
+      .put(Types.DateType.get(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.DATE_TYPE_NAME))
+      .put(Types.TimestampType.withoutZone(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.BIGINT_TYPE_NAME))
+      .put(Types.TimestampType.withZone(), TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.BIGINT_TYPE_NAME))
+      .build();
+
+  public static List<TypeInfo> getColumnTypes(Schema schema) throws Exception {
+    List<Types.NestedField> fields = schema.columns();
+    List<TypeInfo> types = new ArrayList<>(fields.size());
+    for (Types.NestedField field : fields) {
+      types.add(generateTypeInfo(field.type()));
+    }
+    return types;
+  }
+
+  private static TypeInfo generateTypeInfo(Type type) throws Exception {
+    if (primitiveTypeToTypeInfo.containsKey(type)) {
+      return (TypeInfo) primitiveTypeToTypeInfo.get(type);
+    }
+    switch (type.typeId()) {
+      case UUID:
+        return TypeInfoFactory.getPrimitiveTypeInfo(serdeConstants.STRING_TYPE_NAME);
+      case FIXED:
+        return TypeInfoFactory.getPrimitiveTypeInfo("binary");
+      case TIME:
+        return TypeInfoFactory.getPrimitiveTypeInfo("long");
+      case DECIMAL:
+        Types.DecimalType dec = (Types.DecimalType) type;
+        int scale = dec.scale();
+        int precision = dec.precision();
+        try {
+          HiveDecimalUtils.validateParameter(precision, scale);
+        } catch (Exception e) {
+          //TODO Log that precision / scale isn't valid
```

Comment:
Can you add the log message you want now? Or just remove the try/catch as it's not adding anything right now.